packages/rum-core/src/common/patching/event-target-patch.js

```diff
@@ -0,0 +1,185 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import {
+  SCHEDULE,
+  INVOKE,
+  ADD_EVENT_LISTENER_STR,
+  REMOVE_EVENT_LISTENER_STR,
+  EVENT_TARGET
+} from '../constants'
+
+import { apmSymbol } from './patch-utils'
+
+const eventTypes = ['click']
+
+const eventTypeSymbols = {}
+
+for (let i = 0; i < eventTypes.length; i++) {
+  const et = eventTypes[i]
+  eventTypeSymbols[et] = apmSymbol(et)
+}
+
+export function patchEventTarget(callback) {
+  if (!window.EventTarget) {
+    return
+  }
+
+  let proto = window.EventTarget.prototype
+  const nativeAddEventListener = proto[ADD_EVENT_LISTENER_STR]
+  const nativeRemoveEventListener = proto[REMOVE_EVENT_LISTENER_STR]
+
+  function findTaskIndex(existingTasks, eventType, listenerFn, capture) {
+    for (let i = 0; i < existingTasks.length; i++) {
+      const task = existingTasks[i]
+      if (
+        task.eventType === eventType &&
+        task.listenerFn === listenerFn &&
+        task.capture === capture
+      ) {
+        return i
+      }
+    }
+    return -1
+  }
+
+  function isCapture(options) {
+    let capture
+    if (options === undefined) {
+      capture = false
+    } else if (options === true) {
+      capture = true
+    } else if (options === false) {
+      capture = false
+    } else {
+      capture = options ? !!options.capture : false
+    }
+    return capture
+  }
+
+  function createListenerWrapper(target, eventType, listenerFn, options) {
+    const eventSymbol = eventTypeSymbols[eventType]
+
+    if (!eventSymbol) return listenerFn
+
+    let existingTasks = target[eventSymbol]
+
+    let capture = isCapture(options)
+
+    if (existingTasks) {
+      let taskIndex = findTaskIndex(
+        existingTasks,
+        eventType,
+        listenerFn,
+        capture
+      )
+      if (taskIndex !== -1) {
+        let task = existingTasks[taskIndex]
+        return task.wrappingFn
+      }
+    } else {
+      existingTasks = target[eventSymbol] = []
+    }
+
+    let task = {
+      source: EVENT_TARGET,
+      target,
+      eventType,
+      listenerFn,
+      capture,
+      wrappingFn
+    }
+
+    existingTasks.push(task)
+
+    function wrappingFn(event) {
+      task.event = event
```

Comment:
do we need event part of the task? since we already have event type as part of the task.

Reply:
I think the actual event can be used to derive information about it, like the `srcElement` which could be used in the naming the transaction.

Reply:
Can we just pass that instead, reason was it would be hard for the Garbage collection to kick in till the object is being passed around. we can also do this in the coming PR where we will use these tasks and create transaction. What do you think?